### PR TITLE
Feature/main page implementation

### PR DESCRIPTION
**Pull request by daniilandco**

### CODE REVIEW

**Comment by daniilandco** [comment]
Implemented things:

Full project structure refactoring.

Created react components for search bar and displaying requested ads from backend server.

Build with webpack.

Use Eslint and Prettier utilities.

**Comment by daniilandco** [comment]
Feature that is being considered to be implemented next - full functional Nav Bar with authorization.

File: frontend/src/components/Button/Button.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    outline: none;
+    border: none;
+    cursor: pointer;
+}
+
+.btn:hover {
+    padding: 8px 20px;
+    transition: all 0.25s ease-out;
+    background: #fff;
+    color: #d50000;
```
Comment (maryia-kabash) [inline frontend/src/components/Button/Button.css:17]: it's good when all colors go to variables

File: frontend/src/components/Menu/Menu.css [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+.fa-times {
+    color: white;
+}
+
+.menu-icon {
+    display: none;
+}
+
+@media screen and (max-width: 960px) {
+    .NavBarItems {
```
Comment (maryia-kabash) [inline frontend/src/components/Menu/Menu.css:61]: better stick to one naming convention across all styles

File: frontend/src/components/SearchBar/SearchBar.jsx [hunk 3]
```diff
@@ -1,22 +1,35 @@
...
+    const previousValue = useRef('');
+
+    const debouncedResults = useMemo(() => debounce(search.onSearch, 500), []);
 
     const getCurrentValue = () => inputRef.current.value.trim();
 
     useEffect(() => {
-        console.log('search bar render!');
+        previousValue.current = getCurrentValue();
+        return () => debouncedResults.cancel();
```
Comment (maryia-kabash) [inline frontend/src/components/SearchBar/SearchBar.jsx:17]: With no dependencies array this effect will be called on each render

File: frontend/src/modal/RegisterForm.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React, {useContext} from 'react';
+import Button from "../components/Button/Button";
+import useInput from "../hooks/useInput";
+import {Context} from "../index";
+
+const RegisterForm = () => {
+    const {store} = useContext(Context);
+
+    const firstName = useInput('');
```
Comment (maryia-kabash) [inline frontend/src/modal/RegisterForm.jsx:9]: As alternative you can use 'name' for managing inputs state

Comment (maryia-kabash) [reply frontend/src/modal/RegisterForm.jsx:9]: https://reactjs.org/docs/forms.html#handling-multiple-inputs

**Pull request closed without merging**